### ISSUE TITLE
Loss become nan with Adam and AdamW as base optimizers

### ISSUE THREAD

**Issue by FunnyJingl** [pytorch/contrib#37]
Loss becomes nan after training for ~20 steps - loss value stabily decreases and becomes nan with Adam or AdamW optimizers. In case of simple SGD usage it works well.
